src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,29 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.NoSuchElementException;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int FIRST_INDEX = 0;
+    private static final int SECOND_INDEX = 1;
+    private static final int MIN_AGE = 35;
+    private static final int MIN_PERIOD = 10;
+    private static final String NATIONALITY = "Ukrainian";
+    private static final String DELETE_SYMBOL = "-";
+
+    public boolean test(Candidate candidate) {
+        int periodInUkraine = Stream.of(candidate.getPeriodsInUkr())
```

Comment:
create method `checkTimeLivingInCountry(candidate)`

```suggestion
        return candidate.getAge() >= MIN_AGE
                && candidate.isAllowedToVote()
                && candidate.getNationality().equals(NATIONALITY)
                && checkTimeLivingInCountry(candidate);
```